Many tools don't work on aarch64 when kasan is enabled

On aarch64, when kasan is enabled, many tools failed with the following error:
```
# /usr/share/bcc/tools/bashreadline 
In file included from <built-in>:3:
In file included from /virtual/include/bcc/helpers.h:54:
In file included from arch/arm64/include/asm/page.h:44:
arch/arm64/include/asm/memory.h:204:24: error: use of undeclared identifier 'KASAN_SHADOW_SCALE_SHIFT'
        return kimage_vaddr - KIMAGE_VADDR;
                              ^
arch/arm64/include/asm/memory.h:64:24: note: expanded from macro 'KIMAGE_VADDR'
#define KIMAGE_VADDR            (MODULES_END)
                                 ^
arch/arm64/include/asm/memory.h:68:23: note: expanded from macro 'MODULES_END'
#define MODULES_END             (MODULES_VADDR + MODULES_VSIZE)
                                 ^
arch/arm64/include/asm/memory.h:69:25: note: expanded from macro 'MODULES_VADDR'
#define MODULES_VADDR           (BPF_JIT_REGION_END)
                                 ^
arch/arm64/include/asm/memory.h:67:29: note: expanded from macro 'BPF_JIT_REGION_END'
#define BPF_JIT_REGION_END      (BPF_JIT_REGION_START + BPF_JIT_REGION_SIZE)
                                 ^
arch/arm64/include/asm/memory.h:65:42: note: expanded from macro 'BPF_JIT_REGION_START'
#define BPF_JIT_REGION_START    (VA_START + KASAN_SHADOW_SIZE)
                                            ^
arch/arm64/include/asm/memory.h:91:48: note: expanded from macro 'KASAN_SHADOW_SIZE'
#define KASAN_SHADOW_SIZE       (UL(1) << (VA_BITS - KASAN_SHADOW_SCALE_SHIFT))
                                                     ^
1 error generated.
Traceback (most recent call last):
  File "/usr/share/bcc/tools/bashreadline", line 67, in <module>
    b = BPF(text=bpf_text)
  File "/usr/lib/python3.6/site-packages/bcc/__init__.py", line 365, in __init__
    raise Exception("Failed to compile BPF module %s" % (src_file or "<text>"))
Exception: Failed to compile BPF module <text>
```

The reason behind this, is that on arm64 KASAN_SHADOW_SCALE_SHIFT isn't defined in arch/arm64/include/asm/memory.h as ii is for the other arches, but in the arch/arm64/Makefile where it's not visible to bcc.

--- a/src/cc/export/bpf_workaround.h
+++ b/src/cc/export/bpf_workaround.h
@@ -8,4 +8,21 @@ R"********(
 #ifndef __HAVE_BUILTIN_BSWAP64__
 #define __HAVE_BUILTIN_BSWAP64__
 #endif
+
+/**
+ * commit b2f557eae9ed ("kasan, arm64: adjust shadow size for tag-based mode")
+ * KASAN_SHADOW_SCALE_SHIFT moved from headers to the arm64 Makefile
+ * see:
+ *     https://github.com/torvalds/linux/commit/b2f557eae9ed
+ */
+#ifdef __aarch64__
+#if defined(CONFIG_KASAN) && !defined(KASAN_SHADOW_SCALE_SHIFT)
+#ifdef CONFIG_KASAN_SW_TAGS
+#define KASAN_SHADOW_SCALE_SHIFT 4
+#endif
+#ifdef CONFIG_KASAN_GENERIC
+#define KASAN_SHADOW_SCALE_SHIFT 3
+#endif
+#endif
+#endif
 )********"

